src/databricks/labs/ucx/source_code/python_ast_util.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import annotations
+
+import ast
+from dataclasses import dataclass
+
+
+@dataclass
+class Span:
+    """Represents a (possibly multiline) source code span."""
+
+    start_line: int
+    start_col: int
+    end_line: int
+    end_col: int
+
+
+class AstUtil:
+    @staticmethod
+    def extract_callchain(node: ast.AST) -> ast.Call | None:
+        """If 'node' is an assignment or expression, extract its full call-chain (if it has one)"""
+        call = None
+        if isinstance(node, ast.Assign):
+            call = node.value
+        elif isinstance(node, ast.Expr):
+            call = node.value
+        if not isinstance(call, ast.Call):
+            call = None
+        return call
```

Comment:
```suggestion
        if isinstance(node, ast.Assign):
            return node.value
        if isinstance(node, ast.Expr):
            return node.value
        return None
```
init with `None` is often source of bugs. always return early

Reply:
It doesn't seem to be functionally equivalent, is it?
I can split it in two methods instead.